src/Sylius/Bundle/AddressingBundle/Form/Type/CountryChoiceType.php

```diff
@@ -69,6 +74,11 @@ public function configureOptions(OptionsResolver $resolver): void
             ])
             ->setAllowedTypes('choice_filter', ['null', 'callable'])
         ;
+
+        // Don't set the value if there is more than one country because that would override the value that is loadded from the enity
+        if ($defaultValue !== null) {
```

Comment:
Would the following code work instead of using that default value?

```php
$resolver->setDefault('data', function (Options $options, ?CountryInterface $previousCountryOrNull): ?CountryInterface {
    if (count($options['choices']) === 1) {
        return reset($options['choices']);
    }

    return $previousCountryOrNull;
});
```

Reply:
I tried something like this but this overwrote the previously selected country as far as I can remember.